diff --git a/src/NLP/POS/LiteralTagger.hs b/src/NLP/POS/LiteralTagger.hs
index 913bee8..3c2f71d 100644
--- a/src/NLP/POS/LiteralTagger.hs
+++ b/src/NLP/POS/LiteralTagger.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE OverloadedStrings, PackageImports #-}
 module NLP.POS.LiteralTagger
     ( tag
     , tagSentence
@@ -27,7 +27,7 @@ import NLP.FullStop (segment)
 import NLP.Types ( tagUNK, Sentence, TaggedSentence(..), applyTags
                  , Tag, POSTagger(..), CaseSensitive(..), tokens, showTok)
 import Text.Regex.TDFA
-import Text.Regex.TDFA.Text (compile)
+import "regex-tdfa" Text.Regex.TDFA.Text (compile)
 
 taggerID :: ByteString
 taggerID = pack "NLP.POS.LiteralTagger"
