diff --git a/Text/Regex/TDFA/Text.hs b/Text/Regex/TDFA/Text.hs
index c4ef9db..9299272 100644
--- a/Text/Regex/TDFA/Text.hs
+++ b/Text/Regex/TDFA/Text.hs
@@ -38,13 +38,6 @@ import Text.Regex.TDFA.NewDFA.Uncons(Uncons(uncons))
 import qualified Text.Regex.TDFA.NewDFA.Engine as Engine(execMatch)
 import qualified Text.Regex.TDFA.NewDFA.Tester as Tester(matchTest)
 
-instance Extract T.Text where
-  before = T.take; after = T.drop; empty = T.empty
-
-instance Uncons T.Text where
-  {- INLINE uncons #-}
-  uncons = T.uncons
-
 instance RegexContext Regex T.Text T.Text where
   match = polymatch
   matchM = polymatchM
diff --git a/Text/Regex/TDFA/Text/Lazy.hs b/Text/Regex/TDFA/Text/Lazy.hs
index 73ca4a0..52958fb 100644
--- a/Text/Regex/TDFA/Text/Lazy.hs
+++ b/Text/Regex/TDFA/Text/Lazy.hs
@@ -38,17 +38,10 @@ import Text.Regex.TDFA.NewDFA.Uncons(Uncons(uncons))
 import qualified Text.Regex.TDFA.NewDFA.Engine as Engine(execMatch)
 import qualified Text.Regex.TDFA.NewDFA.Tester as Tester(matchTest)
 
-instance Extract L.Text where
-  before = L.take . toEnum; after = L.drop . toEnum; empty = L.empty
-
 instance RegexContext Regex L.Text L.Text where
   match = polymatch
   matchM = polymatchM
 
-instance Uncons L.Text where
-  {- INLINE uncons #-}
-  uncons = L.uncons
-
 instance RegexMaker Regex CompOption ExecOption L.Text where
   makeRegexOptsM c e source = makeRegexOptsM c e (L.unpack source)
 
