From 32a1b9a189c13bec03c6b0f258121c47185db0ad Mon Sep 17 00:00:00 2001
From: Florian Klink <flokli@flokli.de>
Date: Tue, 23 Jan 2024 11:26:10 +0200
Subject: [PATCH] bastardkb dilemma: enable taps

---
 keyboards/bastardkb/dilemma/3x5_3/config.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/keyboards/bastardkb/dilemma/3x5_3/config.h b/keyboards/bastardkb/dilemma/3x5_3/config.h
index ccbc4e2f58..bf17dc7e02 100644
--- a/keyboards/bastardkb/dilemma/3x5_3/config.h
+++ b/keyboards/bastardkb/dilemma/3x5_3/config.h
@@ -42,6 +42,7 @@
 #define POINTING_DEVICE_CS_PIN GP21
 #undef CIRQUE_PINNACLE_DIAMETER_MM
 #define CIRQUE_PINNACLE_DIAMETER_MM 40
+#define CIRQUE_PINNACLE_TAP_ENABLE 1
 
 /* Reset. */
 #define RP2040_BOOTLOADER_DOUBLE_TAP_RESET
-- 
2.43.0

