From 621cadcc1dd71e9397c21cf8cf0f1aae4f6f7057 Mon Sep 17 00:00:00 2001
From: Luke Granger-Brown <git@lukegb.com>
Date: Thu, 2 Jul 2020 23:02:09 +0100
Subject: [PATCH 1/6] Use detzip in download_bower.py

---
 tools/js/download_bower.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/js/download_bower.py b/tools/js/download_bower.py
index d541b565a9..ffdae60f95 100755
--- a/tools/js/download_bower.py
+++ b/tools/js/download_bower.py
@@ -110,7 +110,7 @@ def main():
                 args.b, '--quiet', 'install', '%s#%s' % (args.p, args.v)))
         bc = os.path.join(cwd, 'bower_components')
         subprocess.check_call(
-            ['zip', '-q', '--exclude', '.bower.json', '-r', cached, args.n],
+            ['detzip', '--exclude', '.bower.json', cached, args.n],
             cwd=bc)
 
         if args.s:
-- 
2.32.0

