From 9ed3e8ee2aad704fcfef2490617e2fd22902b2c4 Mon Sep 17 00:00:00 2001
From: Florian Klink <flokli@flokli.de>
Date: Sun, 27 Oct 2024 12:01:13 +0100
Subject: [PATCH] miryoku_behaviors: add quick-tap-ms, require-prior-idle-ms

---
 miryoku/miryoku_behaviors.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/miryoku/miryoku_behaviors.dtsi b/miryoku/miryoku_behaviors.dtsi
index 473df49..73211da 100644
--- a/miryoku/miryoku_behaviors.dtsi
+++ b/miryoku/miryoku_behaviors.dtsi
@@ -14,6 +14,8 @@
       compatible = "zmk,behavior-hold-tap";
       #binding-cells = <2>;
       tapping-term-ms = <U_TAPPING_TERM>;
+      quick-tap-ms = <200>;
+      require-prior-idle-ms = <125>;
       flavor = "tap-preferred";
       bindings = <&mo>, <&kp>;
     };
-- 
2.46.1

